yamllint/cli.py

```diff
@@ -141,6 +141,19 @@ def show_problems(problems, file, args_format, no_warn):
     return max_level
 
 
+def find_project_config_filename(path='.'):
+    for filename in ('.yamllint', '.yamllint.yaml', '.yamllint.yml'):
+        file = os.path.join(path, filename)
+        if os.path.isfile(file):
+            return file
+
+    if os.path.abspath(path) == os.path.abspath(os.path.expanduser('~')):
+        return None
+    if os.path.abspath(path) == os.path.abspath('/'):
+        return None
```

Comment:
> > Will the comparison to `os.path.abspath('/')` work on Windows-like systems?
> 
> It seems that it will work...
> 
> ```
> Python 3.10.8 (tags/v3.10.8:aaaf517, Oct 11 2022, 16:50:30) [MSC v.1933 64 bit (AMD64)] on win32
> Type "help", "copyright", "credits" or "license" for more information.
> >>> import os
> >>> os.getcwd()
> 'C:\\Program Files\\Python310'
> >>> os.path.abspath(os.path.join(os.getcwd(), '..', '..'))
> 'C:\\'
> >>> os.path.abspath('/')
> 'C:\\'
> >>> os.path.abspath(os.path.join(os.getcwd(), '..', '..')) == os.path.abspath('/')
> True
> ```

OK so it works for `C:\\`. But what if the project is under `D:\\`? I fear a possible infinite loop in some situations.

Maybe a safer implementation would be to check whether `os.path.abspath(path) == os.path.abspath(os.path.join(path, '..'))`, and stop if it's the case?

Reply:
You're spot on. Corrected, squashed, and pushed.